### DISCUSSION TITLE
Getting News using APIs

### DISCUSSION THREAD

**Discussion by Hajelarohit** [Budibase/budibase#6093] [Help]
Hi BB team
I am trying to get a newsfeed to my app via APIs. The APIs are connecting correctly and displaying JSON data, but i am unable to pull it into a table form.

What am i doing wrong?

https://vimeo.com/714525782/15cf272b3c

**Comment by shogunpurple** [comment]
Hi @Hajelarohit - you need to map over the response to get it into a format that budibase expects. In this case it's fairly simple, you just need to grab the `articles` key in your tranformer.

```js
return data.articles
```

This should give you the data back!

**Comment by Hajelarohit** [comment]
That worked like a charm.

follow up questions..
1. How can i integrate multiple API queries into a single table?
2. how do i display the information on /:id  page when user clicks on a row. Say i want to read the new article in row 3.
3. How can I add additional field/column? For example i want to provide user an option to filter news by supplier name. So how can i add a column in the table and add supplier name by default. Since i created a single API for a single supplier, i can just hard code the supplier name to that query and integrate with other news queries so that all news appears in a single table and user has the option to seach by supplier name.

**Comment by Hajelarohit** [comment]
I kind of found a band aid solution, but let me know if there is a dynamic way.

I created nested data providers and put a table at the bottom and a option picker. i assigned each data providers to a nest query. I told option picker to pick the vendor name from another table and told table condition that when option picker picks a vendor name, change the data provider to the one that has the respective nest query.

This is okay if there are only 2-5 nested data providers for each supplier, but it will require a lot of manual work if there are more than 1000 suppliers. let me know if there is a better way. Also need answers to my previous questions
